redbot/cogs/general/general.py

```diff
@@ -220,31 +220,164 @@ def __init__(self):
 
     @commands.command()
     @commands.guild_only()
-    async def serverinfo(self, ctx):
-        """Show server information."""
+    async def serverinfo(self, ctx, details: bool = False):
+        """
+            Show server information.
+        
+            `details`: Toggle it to `True` to show more
+            information about this server.
+            Defaults to False.
+        """
         guild = ctx.guild
-        online = len([m.status for m in guild.members if m.status != discord.Status.offline])
+
+        def check_feature(feature):
+            return "\N{WHITE HEAVY CHECK MARK}" if feature in guild.features else "\N{CROSS MARK}"
+
+        format_kwargs = {
+            "vip": check_feature("VIP_REGIONS"),
+            "van": check_feature("VANITY_URL"),
+            "splash": check_feature("INVITE_SPLASH"),
+            "m_emojis": check_feature("MORE_EMOJI"),
+            "verify": check_feature("VERIFIED"),
+        }
+
+        verif = {
+            0: _("0 - None"),
+            1: _("1 - Low"),
+            2: _("2 - Medium"),
+            3: _("3 - Hard"),
+            4: _("4 - Extreme"),
+        }
+        region = {
+            "vip-us-east": _("__VIP__ US East") + " :flag_us:",
+            "vip-us-west": _("__VIP__ US West") + " :flag_us:",
+            "vip-amsterdam": _("__VIP__ Amsterdam") + " :flag_nl:",
+            "eu-west": _("EU West") + " :flag_eu:",
+            "eu-central": _("EU Central") + " :flag_eu:",
+            "london": _("London") + " :flag_gb:",
+            "frankfurt": _("Frankfurt") + " :flag_de:",
+            "amsterdam": _("Amsterdam") + " :flag_nl:",
+            "us-west": _("US West") + " :flag_us:",
+            "us-east": _("US East") + " :flag_us:",
+            "us-south": _("US South") + " :flag_us:",
+            "us-central": _("US Central") + " :flag_us:",
+            "singapore": _("Singapore") + " :flag_sg:",
+            "sydney": _("Sydney") + " :flag_au:",
+            "brazil": _("Brazil") + " :flag_br:",
+            "hongkong": _("Hong Kong") + " :flag_hk:",
+            "russia": _("Russia") + " :flag_ru:",
+            "japan": _("Japan") + " :flag_jp:",
+            "southafrica": _("South Africa") + " :flag_za:",
+            "india": _("India") + " :flag_in:",
+        }
+
+        online = len([m.status for m in guild.members if m.status == discord.Status.online])
+        idle = len([m.status for m in guild.members if m.status == discord.Status.idle])
+        dnd = len([m.status for m in guild.members if m.status == discord.Status.dnd])
+        offline = len([m.status for m in guild.members if m.status == discord.Status.offline])
+        streaming = len([m for m in guild.members if isinstance(m.activity, discord.Streaming)])
+        mobile = len([m for m in guild.members if m.is_on_mobile()])
+        lurkers = len([m for m in guild.members if m.joined_at is None])
```

Comment:
Once this line is removed (and further uses of the var are gone) this is still a change we can add if you're still interested in doing this. (the lurkers being exposed like this was an API bug when it was happening (they aren't anymore) and there are other cases where joined_at can be None)

Reply:
I just updated the PR with all changes that I've done since last time I touched that PR.